dashboard/app/models/concerns/serialized_to_file_validation.rb

```diff
@@ -0,0 +1,16 @@
+# This concern validates name for cases where name is used to serialize a model
+# to the filesystem
+module SerializedToFileValidation
+  extend ActiveSupport::Concern
+
+  included do
+    validates :name,
+      presence: true,
+      uniqueness: {case_sensitive: false},
+      format: {
+        unless: :skip_name_format_validation,
```

Comment:
Will this blow up if `skip_name_format_validation` is not implemented in the model?

We could be a bit friendlier, something like?:
```ruby
unless -> {respond_to? :skip_name_format_validation && skip_name_format_validation}
```